api/insight/socket.io.go

```diff
@@ -100,54 +103,86 @@ func NewSocketServer(params *chaincfg.Params, txGetter txhelpers.RawTransactionG
 		PingTimeout:  5 * time.Second,
 		Transports:   []transport.Transport{wsTrans},
 	}
-	server, err := socketio.NewServer(opts)
+	socketIOServer, err := socketio.NewServer(opts)
 	if err != nil {
 		apiLog.Errorf("Could not create socket.io server: %v", err)
 		return nil, err
 	}
 
-	// Set maximum number of connections.
-	//server.SetMaxConnection(16384)
-
 	// Each address subscription uses its own room, which has the same name as
 	// the address. The number of subscribers for each room is tracked.
-	addrs := roomSubscriptionCounter{
-		RWMutex: new(sync.RWMutex),
-		c:       make(map[string]int),
+	addrs := &roomSubscriptionCounter{
+		c: make(map[string]int),
+	}
+
+	server := &SocketServer{
+		Server:           socketIOServer,
+		params:           params,
+		watchedAddresses: addrs,
+		txGetter:         txGetter,
 	}
 
+	// OnConnect sets the address room subscription counter to 0. There are no
+	// default subscriptions. The client must subscribe to "inv" if they want
+	// notification of all new transactions.
 	server.OnConnect("/", func(so socketio.Conn) error {
-		// New connections automatically join the inv and sync rooms.
-		so.Join("inv")
-		so.Join("sync")
-		apiLog.Debugf("New socket.io connection. %d clients are connected.",
-			server.RoomLen("inv"))
+		so.SetContext(uint32(0))
```

Comment:
Yeah, I'll document that since it's not clear until you get to the subscribe event handler.  That's the idea though, it's a general purpose connection-scoped data element.